backend/typescript/services/implementations/reviewService.ts

```diff
@@ -265,6 +301,20 @@ class ReviewService implements IReviewService {
           }
         });
 
+        // Delete genres (if necessary)
+        book.genres.forEach(async (genre: PgGenre) => {
```

Comment:
Just tested this. Looks like this block isn't actually deleting the genre when no books are associated with it. We ran into a similar issue with Tags (See #136), so you can just copy that fix (forewarned, it's kinda ugly :cry:).

I think this has to do with async weirdness but not sure :shrug: